File: logistics-realtime/src/config/env.js
import dotenv from "dotenv";

dotenv.config();

const get = (key, fallback) => {
  const val = process.env[key];
  if (val === undefined || val === "") return fallback;
  return val;
};

const env = {
  PORT: Number(get("PORT", 9000)),
  NODE_ENV: get("NODE_ENV", "development"),
  REDIS_URL: get("REDIS_URL", "redis://127.0.0.1:6379"),
  JWT_SECRET: get("JWT_SECRET"),
  CORS_ORIGIN: get("CORS_ORIGIN", "*"),
  HEARTBEAT_TTL: Number(get("HEARTBEAT_TTL", 30)),
  LOCATION_UPDATE_INTERVAL_MS: Number(get("LOCATION_UPDATE_INTERVAL_MS", 5000)),
};

if (!env.JWT_SECRET) {
  console.warn(
    "Warning: JWT_SECRET is not set. Socket auth will fail without a secret in production."
  );
}

export default env;

File: rust-liveTracking-server/src/config/env.rs
use std::env;

pub struct Env {
	pub port: u16,
	pub redis_url: String,
	pub jwt_secret: String,
	pub cors_origin: Option<String>,
}

impl Env {
	pub fn from_env() -> Self {
		let port = env::var("PORT").ok().and_then(|s| s.parse().ok()).unwrap_or(8080);
		let redis_url = env::var("REDIS_URL").unwrap_or_else(|_| "redis://127.0.0.1/".into());
		let jwt_secret = env::var("JWT_SECRET").unwrap_or_else(|_| "secret".into());
		let cors_origin = env::var("CORS_ORIGIN").ok();
		Env { port, redis_url, jwt_secret, cors_origin }
	}
}
